packages/lib/isOutOfBounds.tsx

```diff
@@ -219,19 +240,30 @@ export function isTimeViolatingFutureLimit({
   periodLimits: PeriodLimits;
 }) {
   const log = logger.getSubLogger({ prefix: ["isTimeViolatingFutureLimit"] });
-  const date = dayjs(time);
-  if (periodLimits.rollingEndDay) {
-    const isAfterRollingEndDay = date.isAfter(periodLimits.rollingEndDay);
-    log.debug({
-      formattedDate: date.format(),
+  // Because we are just going to compare times, we need not convert the time to a particular timezone
+  // date.isAfter/isBefore are timezone neutral
+  const dateInSystemTz = dayjs(time);
```

Comment:
Just renamed `date` to convey the tz and added comment as to why the timezone doesn't matter in this function